src/sdk/auth/browser-auth-provider.ts

```diff
@@ -93,6 +117,15 @@ export class IlandBrowserAuthProvider implements AuthProvider {
       });
     });
   }
+
+  /**
+   * If we have an impersonated role initialized with the SDK, we need to update all token.
+   * @param token
+   * @returns {string} the complete token.
+   */
+  private getImpersonateToken(token?: string): string {
+    return this._testingRoleUuid ? `impersonate=${this._testingRoleUuid};${token || ''}` : token || '';
```

Comment:
if token is falsy, you should just return token right away vs. using the || ''

Reply:
ho ok. so the function can return a `string` or `undefined` actually right ?

Reply:
yeah if the token itself can be undefined then the result of this function needs to be able to be undefined as well

Reply:
Alright. it's now the case ;-). i've just updated the PR.